File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
a, b = map(int,input().split())

# 최대공약수
def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a

# 최소공배수
def lcm(a, b):
    return (a * b) // gcd(a, b)

print(gcd(a, b)) # 최대공약수
print(lcm(a, b)) # 최소공배수

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split(' ');
const [a, b] = input.map(Number);

// 최대공약수 구하기 (유클리드 호제법)
function gcd(x, y) {
  while (y !== 0) {
    let temp = y;
    y = x % y;
    x = temp;
  }
  return x;
}

// 최소공배수 구하기
function lcm(x, y) {
  return (x * y) / gcd(x, y);
}

const gcdResult = gcd(a, b);
const lcmResult = lcm(a, b);

console.log(gcdResult);
console.log(lcmResult);
